### DISCUSSION TITLE
Quadratic Program with Opti stack

### DISCUSSION THREAD

**Discussion by LoreMoretti** [casadi/casadi#3375] [Q&A]
Hello,

I know that with the Opti stack we can formulate generic NLP problems, and solve them for example leveraging "ipopt".

I was wondering if there is a way to formulate a **QP problem** with the **Opti stack**, and solve it with a **dedicated solver** such as "osqp", rather than deploying "ipopt".

In some early experiments, I have formulated the QP problem with the Opti stack, and deployed as a solver `opti.solver("sqpmethod", {"qpsol": "osqp"})`. I have noticed a speed up if compared to "ipopt".

But is this the correct way to handle QP problems with Opti?

Thanks a lot!

Lorenzo

**Comment by jgillis** [comment]
Use `Opti('conic')`

**Reply by LoreMoretti** [reply]
Hi @jgillis 

thanks a lot for your super fast reply! It works :)
